keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -1172,6 +1207,179 @@ back to aggregating container requests.
   size of the pod's cgroup. This means the pod cgroup's resource limits will be
   set to accommodate both pod-level requests and pod overhead.
 
+#### Hugepages
+
+With the proposed changes, hugepages-2Mi and hugepages-1Gi will be added to the pod-level resources section, alongside CPU and memory. The hugetlb cgroup for the
+pod will then directly reflect the pod-level hugepage limits, rather than using an aggregated value from container limits. When scheduling, the scheduler will 
+consider hugepage requests at the pod level to find nodes with enough available
+resources.
+
+Containers will still need to mount an emptyDir volume to access the huge page filesystem (typically /dev/hugepages).  This is the standard way for containers to interact with huge pages, and this will not change. 
+
+#### Memory Manager
+
+With the introduction of pod-level resource specifications, the Kubernetes Memory
+Manager will evolve to track and enforce resource limits at both the pod and
+container levels. It will need to aggregate memory usage across all containers
+within a pod to calculate the pod's total memory consumption. The Memory Manager
+will then enforce the pod-level limit as the hard cap for the entire pod's memory
+usage, preventing it from exceeding the allocated amount.  While still
+maintaining container-level limit enforcement, the Memory Manager will need to
+coordinate with the Kubelet and eviction manager to make decisions about pod
+eviction or individual container termination when the pod-level limit is
+breached.
+
+#### CPU Manager
+
+With the introduction of pod-level resource specifications, the CPU manager in
+Kubernetes will adapt to manage CPU requests and limits at the pod level rather
+than solely at the container level. This change means that the CPU manager will
+allocate and enforce CPU resources based on the total requirements of the entire
+pod, allowing for more flexible and efficient CPU utilization across all
+containers within a pod. The CPU manager will need to ensure that the aggregate
+CPU usage of all containers in a pod does not exceed the pod-level limits. 
```

Comment:
what about the exclusive allocation of CPUs? Is that something we keep at container level?

Reply:
Thanks for the questions @swatisehgal. Scoping out CPU manager also for Beta then as it requires thinking about all the different policy options before finalizing the design. The main focus of alpha 2 in 1.33 is in-place pod related API changes so removing other complex features out of scope of alpha 2 then. I will revisit these for Beta after discussions offline and in sig node meetings.

Reply:
I'm late to the party, but @swatisehgal captured the most pressing points anyway. Looking forward to join the future conversations in sig-nodel.